manifestoes/extensions.rst

```diff
@@ -0,0 +1,1675 @@
+      every use of a rebindable construct that does not refer to the
+      appropriate name from base.
+
+   -  ``-XWarnDeepSubsumption``: would warn when deep subsumption was
+      actually used, and simple subsumption would not have sufficed.
+
+The meaning of ``-W`` and ``-Wall`` would continue to be "enable all recommended
+warnings" and "enable all reasonable warnings", just as in GHC today.
+These lists may vary with GHC version and language edition.
+
+Consequences and benefits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  The tension between warnings and language extensions disappears.
+   For example, at the top of a module we can write ::
+
+     {-# LANGUAGE GADTs, NoIncompletePatterns #-}
+
+   rather than ::
+
+     {-# LANGUAGE GADTs #-}
+     {-# OPTIONS_GHC -Werror=incomplete-patterns #-}
+
+-  A language edition could choose to allow, but warn, about a language
+   extension, e.g ``-XDeriveFunctor``. That is not possible today.
+
+Questions
+~~~~~~~~~~~~~
+
+-  How does this play with the new user defined warning categories?
+
+-  A new language extension for each warning, and a new warning for each
+   language extension. Two long lists (extensions and warnings) combined
+   into one even longer list. Could feel intimidating.
+
+-  Will we end up supporting something for longer? Eg ``-Wmonad-fail``. It
+   lived only for a few releases, it warned you if you didn't write your
+   code in a forward compatible way. E.g ``-Wstar-is-type``. The language
+   extension ``-XStarIsType`` could go away entirely.
+
+.. [#]
+   The semantics also come with compilation outputs, but we’ll ignore
+   that important detail in this document.
+
+.. [#]
+   Of course, Haskell isn’t completely pure, supporting unsafePerformIO
+   and friends. The behavior of such functions is often important in
+   Haskell programs. In thinking about stability, we should be attuned
+   to possible changes here. If we identify any particular patterns of
+   impure functions that we wish to guarantee future support for, we
+   should document them.
+
+.. [#]
+   With any of the new language editions specified (that is, anything
+   other than ``Haskell98``, ``Haskell2010``, or ``GHC2021``),
+   ``ExistentialQuantification`` will allow only existential quantification,
+   not also (and accidentally) GADTs.
+
+.. [#]
+   ``LexicalNegation`` fixes a wart in the language around unary ``-``, by
+   requiring unary ``-`` not to have any whitespace after it. However, unary
+   ``-`` remains the only symbolic unary function in the language. I hereby
+   propose fixing the wart in the other direction, by just killing off unary
+   ``-``. In the presence of ``NegativeLiterals``, we can still write ``-5``
+   for negative 5. But if you want to negate ``x``, just ``negate x``.
+
+   Thus with any of the new language editions, users cannot write a unary
+   ``-``, unless ``Classic`` mode is on.
+
+.. [#]
+   It is time for the ``Safe`` ecosystem to be dismantled. We should warn
```

Comment:
I agree with this, but I was surprised to see this buried in a footnote! For a while I've felt that `Safe` is not really pulling its weight, but proposing to remove it seemed like it risked sparking too much controversy. I think this proposal provides a good opportunity to argue the case for removal as part of the broader shake-up.

Reply:
> I've felt that Safe is not really pulling its weight, but proposing to remove it seemed like it risked sparking too much controversy

I understood the outcome of https://discourse.haskell.org/t/deprecating-safe-haskell-or-heavily-investing-in-it/5489 to be that `Safe` was considered due for removal by a motivated person. N.B. @simonpj's comment

> I think it would be plausible to lay it to rest

Reply:
In a fit of enthusiasm a few years ago, I almost removed Safe. What stayed my hand is that it's goals are worthy, and any replacement of Safe will have to have a similar community buy-in and trust network. So it seemed a shame to disrupt the existing trust network just to have to rebuild it. For whatever reason, I'm now more confident that a replacement isn't coming anytime soon and that it really is time to leave Safe behind.